src/kraken/core/address/_address_resolver.py

```diff
@@ -220,8 +220,17 @@ def recurse_tree(entity: T_Addressable, include_root: bool = True) -> Iterable[T
         for child in space.get_children(entity):
             yield from recurse_tree(child)
 
+    def has_children(
+        space: AddressSpace[T_Addressable],
+        addr: T_Addressable,
+    ) -> bool:
+        return len(list(space.get_children(addr))) > 0
```

Comment:
nit (it avoids casting to a list, even if the impact is not huge)
```suggestion
        return any(True for _ in iterator)
```